spec/models/comment_spec.rb

```diff
@@ -1,6 +1,16 @@
 require 'rails_helper'
 
 RSpec.describe Comment, type: :model do
+
+  it { should belong_to(:user) }
+  it { should belong_to(:parent).class_name(:Comment).with_foreign_key(:parent_comment_id) }
+  it { should have_many(:children).class_name(:Comment).with_foreign_key(:parent_comment_id) }
+
+  it { should validate_presence_of :text }
+  it { should validate_presence_of :user_id }
+  it { should validate_presence_of :link_id }
+  it { should validate_presence_of :votes_count }
```

Comment:
will `votes_count` eventually be a [counter cache](http://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-counter-cache)?